app/views/products/_product_form.html.erb

```diff
@@ -0,0 +1,23 @@
+<section>
+  <h3><%= page_title %></h3>
+  <section>
+    <%= form_for @product do |f| %>
+    <%= f.label :product_name %>
+    <%= f.text_field :name %>
+    <%= f.label :unit_price %>
+    <%= f.text_field :price_from_form %>
```

Comment:
I'm pretty surprised you used `f.text_field` here even though on the same page you have `f.number_field` (line 12)? You could have had it accept decimals by adding `, step: :any`